include/eigenpy/registration_class.hpp

```diff
@@ -30,6 +30,12 @@ class registration_class {
     return *this;
   }
 
+  template <class DerivedVisitor>
+  self& def(bp::def_visitor<DerivedVisitor> const& visitor) {
+    static_cast<DerivedVisitor const&>(visitor).visit(*this);
```

Comment:
Is the downcast actually required? I thought boost::python::def_visitor was a crtp base class and already did the downcast.

Reply:
it does, see "boost/python/def_visitor.hpp" line 78

Reply:
Unfortunatly, the method `DerivedVisitor const& def_visitor<DerivedVisitor>::derived_visitor() const` is protected.

And ` void def_visitor<DerivedVisitor>::visit(classT& c) const` is private :(

Reply:
Oh I think you told me, why would they even do that...